### PR TITLE
Github Stream

### LINKED ISSUE #95: As an author, I want to pull in my github activity to my “stream”

### PR DESCRIPTION
**Pull request by maharshmellow**
Fixes #95 
To run locally, add this to your .env file: 
`github_token=91b1ebc4b420a446140dd89d16e97eb0b199dae6`

and make sure that your user object contains your Github URL (ex: https://github.com/maharshmellow) 

---
- loads posts from the Github API
- merges them with our regular post data 
- normally this takes around 1.5 seconds, but I added a cache that keeps the github data for 300s so it doesn't need to keep getting it - lowers the response time to around 200ms for subsequent API calls
- renders the posts on the homepage with a new GithubPost component

### CODE REVIEW

File: backend/helpers/github.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            "content": message,
+            "title": "Github",
+            "visibility": "PRIVATE",
+            "published": event["created_at"].replace("Z", ".000000Z"),
+            "comments": [],
+            "isGithubPost": True, # used to tell React to render this post differently compared to other Posts
+        }
+
+        parsed_events.append(parsed_event)
+        # TODO: need to add more event types but we can come back to that later
```
Comment (maharshmellow) [inline backend/helpers/github.py:87]: Currently it doesn't handle all the event types - we will need to come back to this at some point to get them all added in but I added the major ones.

File: backend/helpers/github.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    return parsed_events
+
+def load_github_events(github_url, access_token):
+    try:
```
Comment (maharshmellow) [inline backend/helpers/github.py:92]: got lazy and didn't want to error check every single component 😞

File: backend/helpers/github.py (hunk 1 above)
Comment (zhaoyuxuan) [reply backend/helpers/github.py:87]: sad..

**Review by zhaoyuxuan — DISMISSED** [review]
fix the conflict, and it will be good to go

**Review by sandysaji — DISMISSED** [review]
Looks good!

**Review by ceciliaccwei — APPROVED** [review]
Perfect😄👍💯

**Pull request merged**